api/clients/codecs/blob_codec.go

```diff
@@ -4,22 +4,22 @@ import (
 	"fmt"
 )
 
-type BlobEncodingVersion byte
+type PayloadEncodingVersion byte
```

Comment:
```suggestion
type PayloadEncodingVersion uint8
```

Reply:
byte and uint8 are aliases afaik. uint8 feels more appropriate here imo

Reply:
Also should Payload related stuff be in its own codec file? (not sure, just asking)